libs/librrgraph/src/base/rr_spatial_lookup.h

```diff
@@ -220,17 +231,18 @@ class RRSpatialLookup {
   private:
     /* An internal API to find all the nodes in a specific location with a given type
      * For OPIN/IPIN nodes that may exist on multiple sides, a specific side must be provided  
-     * This API is NOT public because its too powerful for developers with very limited sanity checks 
+     * This API is NOT public because it is too powerful for developers with very limited sanity checks
      * But it is used to build the public APIs find_channel_nodes() etc., where sufficient sanity checks are applied
      */
-    std::vector<RRNodeId> find_nodes(int x,
+    std::vector<RRNodeId> find_nodes(int layer,
+                                     int x,
                                      int y,
                                      t_rr_type type,
                                      e_side side = SIDES[0]) const;
 
     /* -- Internal data storage -- */
   private:
-    /* Fast look-up: TODO: Should rework the data type. Currently it is based on a 3-dimensional arrqay mater where some dimensions must always be accessed with a specific index. Such limitation should be overcome */
+    /* Fast look-up: TODO: Should rework the data type. Currently it is based on a 3-dimensional array mater where some dimensions must always be accessed with a specific index. Such limitation should be overcome */
```

Comment:
3D array --> 4D matrix
(I think)

(Why didn't we make this whole thing a 5D matrix instead of an array of 4D matrices then?)  TODO: could try that.